test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,459 @@
+package e2e
+
+import (
+	crdv1beta1 "antrea.io/antrea/pkg/apis/crd/v1beta1"
+	agentconfig "antrea.io/antrea/pkg/config/agent"
+	"antrea.io/antrea/test/e2e/utils"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	v12 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+	"testing"
+	"time"
+)
+
+/*
+1) Create the KIND cluster.
+2) Once cluster is up , create a  service.
+3) Get the IP of above service and configure the same in antrea configMap.
+4) Update antrea configMap.
+
+5) Create NGINX deployment.
+6) Get IP of one of the pods of nginx.
+7) create and configure the custom CoreDNS configMap with the IP received above.
+8) Create custom CoreDNS deployment.
+9) Create and apply antrea FQDN policy.
+10) Deploy antrea-toolbox.
+
+
+---------- tic
+11) curl the FQDN from within toolbox.
+12) imitate caching the IP belonging to above FQDN resolution by keeping it in a variable.
+13) edit configmap with the other IP.
+14) wait for new IP to get updated in configMap and let the changes be reflected in dns pod.
+15) curl the FQDN again with IP , simulating usage of cache -- and it must fail with no connectivity.
+*/
+
+func TestFQDNPolicyWithCachedDNS(t *testing.T) {
+	skipIfAntreaPolicyDisabled(t)
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+
+	testFqdn := "nginx-test-pod.lfx.test"
+
+	//TODO: Check for IPv6 ?
+	ipFamily := v1.IPv4Protocol
+
+	// Create the service .
+	//TODO: Should the names be put up as constants instead of direct strings here?
+	customDnsService, err := testData.CreateUDPService("custom-dns-service", data.testNamespace, 53, 53, map[string]string{"app": "custom-dns"}, false, false, v1.ServiceTypeClusterIP, &ipFamily)
+	if err != nil {
```

Comment:
I think `require.NoError` will already call `T.FailNow`, so all error message you have in the `Fatalf` can be put into the `require.NoError` argument list. This way error handling from here and down below can be much cleaner